sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java

```diff
@@ -167,6 +171,17 @@ public String toString() {
      * from the current transform or component.
      */
     ItemBuilder add(String key, Class<?> value);
+
+    /**
+     * Register the given display metadata. The input value will be inspected to see if it conforms
+     * to one of the supported DisplayData types. Otherwise, it will be registered as a
+     * {@link DisplayData.Type#STRING}, using the {@link Object#toString()} method to retrieve the
+     * display value.
+     *
+     * <p> The added display data is identified by the specified key and namespace from the current
+     * transform or component.
+     */
+    ItemBuilder add(String key, Object value);
```

Comment:
Sounds like an interesting way to register paths.

What bugs me about this overload right now is that it does some "magic", but the other overloads are very safe/explicit. For example, if we had the following code:

    class MyTransform /* extends ... */ {
        private Instant startTime;
        @Override public void populateDisplayData(Builder builder) { 
            builder.add("startTime", startTime); 
        }
    }

If the transform author refactors the type of `startTime`, say to a `java.util.Date`, the code would still compile but would register an unexpectedly different display data type.

Another option is to give  this overload a different name to distinguish it. `addDynamic()`, `tryAdd()`. Or simply `doMagic()`.